packages/protocol/contracts/bridge/Bridge.sol

```diff
@@ -228,52 +230,55 @@ contract Bridge is EssentialContract, IBridge {
 
         bytes32 msgHash = hashMessage(_message);
         _checkStatus(msgHash, Status.NEW);
-        _consumeEtherQuota(_message.value + _message.fee);
 
         address signalService = resolve(LibStrings.B_SIGNAL_SERVICE, false);
 
         ProcessingStats memory stats;
+        stats.proofSize = uint32(_proof.length);
         stats.numCacheOps =
             _proveSignalReceived(signalService, msgHash, _message.srcChainId, _proof);
 
-        uint256 refundAmount;
-        if (_unableToInvokeMessageCall(_message, signalService)) {
-            // Handle special addresses that don't require actual invocation but
-            // mark message as DONE
-            refundAmount = _message.value;
-            _updateMessageStatus(msgHash, Status.DONE);
+        if (!_consumeEtherQuota(_message.value + _message.fee)) {
+            stats.status = Status.RETRIABLE;
```

Comment:
Maybe we should also only allow the owner to make it retriable? Even though whoever processes the message doesn't get any fees so normally it won't happen anyway, but making it retriable would be quite inconvenient so maybe can't hurt to protect against it.

Reply:
Done. please take another look.